x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -57,12 +59,45 @@
 import java.util.stream.Collectors;
 
 import static org.elasticsearch.index.query.QueryBuilders.boolQuery;
+import static org.elasticsearch.xpack.ql.index.IndexResolver.UNMAPPED;
 import static org.elasticsearch.xpack.ql.util.ActionListeners.map;
 import static org.elasticsearch.xpack.ql.util.StringUtils.WILDCARD;
 
 public class EsqlSession {
 
     private static final Logger LOGGER = LogManager.getLogger(EsqlSession.class);
+    public static final BiFunction<String, Map<String, FieldCapabilities>, InvalidMappedField> SPECIFIC_VALIDITY_VERIFIER = (
```

Comment:
I think this might be more clear as a static method and then you do a `EsqlSession::specificValidity` or something.